contracts/accessories/Multicall.vy

```diff
@@ -17,7 +17,7 @@ struct Call:
 @external
 def execute(calls: DynArray[Call, 100]):
     # NOTE: Can only work in a EIP-7702 context from Purse
-    assert msg.sender == self, "Multicall:!authorized"
+    assert tx.origin == self, "Multicall:!authorized"
```

Comment:
I think this might be okay, and allows making another multicall deeper within the callframe stack if you want it (while still restricting call convention to be only when you explicitly make the call)